services/ark-api/ark-api/src/ark_api/core/namespace.py

```diff
@@ -0,0 +1,55 @@
+"""Namespace detection utilities."""
+import logging
+import os
+from pathlib import Path
+
+logger = logging.getLogger(__name__)
+
+# Global variable to store the current namespace
+_current_namespace: str = "default"
+
+
+def is_namespace_detection_enabled() -> bool:
+    """Check if namespace detection is enabled via environment variable."""
+    env_value = os.getenv("ARK_DETECT_NAMESPACE", "false")
+    enabled = env_value.lower() == "true"
+    logger.info(f"ARK_DETECT_NAMESPACE={env_value}, namespace detection enabled: {enabled}")
+    return enabled
+
+
+def detect_current_namespace() -> str:
+    """
+    Detect the current namespace by reading the service account namespace file.
+    Falls back to 'default' if the file is not found or cannot be read.
+    If ARK_DETECT_NAMESPACE is false, always returns 'default'.
+    """
+    # If namespace detection is disabled, always return default
+    if not is_namespace_detection_enabled():
+        logger.info("Namespace detection disabled via ARK_DETECT_NAMESPACE, using default")
+        return "default"
+
+    namespace_file = Path("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
```

Comment:
as per above this can probably move into `ark-sdk` so that it uses this file, falls back to `.kube` context and falls back to default